ArduCopter/takeoff.cpp

```diff
@@ -55,6 +55,15 @@ bool Copter::do_user_takeoff(float takeoff_alt_cm, bool must_navigate)
     return false;
 }
 
+bool Copter::do_user_go_around(float go_around_alt_cm) 
+{
+    if (motors->armed() && control_mode == LAND && set_mode(LOITER, MODE_REASON_GCS_COMMAND)) {
```

Comment:
instead of goind loiter we should go back to the last mode, otherwise coming from guided to loiter without radio, will only crash the drone !

Reply:
Yes, you are right. Thanks a lot for remind me this important point.

Reply:
I think we should only allow go_around if last mode before land is auto, guided, loiter or poshold. If last mode before land is auto, then we should go to guided?

Reply:
It should probably be allowed regardless of the last mode. But it needs to be smart about it it.

If the last mode was guided, it should go back into guided since that's most likely what the operator needs. If the last mode was any other mode, it should probably go into loiter where it can await an operator command.  That puts it inline with what happens when you use the landing-abort that already exists on the throttle stick.

Reply:
pretty please use meters when using floats. Centimeters are for integers. We should try and use SI units (meters) wherever possible.

Reply:
for readability, having these on three lines is easier to read

Reply:
but Copter::takeoff_timer_start accept alt_cm as float

Reply:
correct. So use int when you have to, don't otherwise. Just because one function uses cm, doesn't mean the rest should. Long-term we convert everything to float for standard units.

Reply:
OK. I got it. thank you